### PR TITLE
fix: ensure we properly parse IAM_ROLE that is in ARN format

### PR DESCRIPTION
**Pull request by josegonzalez**
- reformat IAM_ROLE when it matches ARN format
- add logging for when the expected role does not match the available role
- added PYTHONUNBUFFERED so logs come out as they are made available vs when python decides its time
- send log-file to stdout as well

I have signed the CLA.

### CODE REVIEW

File: metadataproxy/roles.py [hunk 1]
```diff
@@ -155,9 +156,10 @@ def find_container(ip):
     return None
 
 
-def check_role_name_from_ip(ip, requested_role):
-    role_name = get_role_name_from_ip(ip)
-    if role_name == requested_role:
+def check_role_name_from_ip(ip, requested_role, expected_role_name=None):
```
Comment (ryan-lane) [inline metadataproxy/roles.py]: I think the previous logic here was ok and matches the name of the function well. Can we keep this logic and move the logging line from the route into this location?

Comment (josegonzalez) [inline metadataproxy/roles.py]: Sounds good.

**Comment by ryan-lane** [comment]
logic also needs to be added for the route in mock.py

**Comment by ryan-lane** [comment]
+1 this is great. thanks for the contribution!

File: metadataproxy/roles.py [hunk 2]
```diff
@@ -158,6 +159,9 @@ def find_container(ip):
 def check_role_name_from_ip(ip, requested_role):
     role_name = get_role_name_from_ip(ip)
     if role_name == requested_role:
+        log.debug('Expected Role: {0}, Actual Role: {}'.format(
+            expected_role_name, requested_role
```
Comment (ryan-lane) [inline metadataproxy/roles.py]: I think this should be:

```
        log.debug('Expected Role: {0}, Actual Role: {1}'.format(
            requested_role, role_name
        ))
```

**Comment by ryan-lane** [comment]
Released as 1.3.0.

**Pull request merged**